src/Lean/Server/FileWorker.lean

```diff
@@ -143,17 +146,20 @@ section ServerM
       snap
     | Sum.inr msgLog =>
       sendDiagnostics msgLog
-      throw TaskError.eof
+      throw ElabTaskError.eof
 
   /-- Elaborates all commands after `initSnap`, emitting the diagnostics. -/
-  def unfoldCmdSnaps (m : DocumentMeta) (initSnap : Snapshot) (cancelTk : CancelToken) : ServerM (AsyncList TaskError Snapshot) := do
-    -- TODO(MH): check for interrupt with increased precision
-    AsyncList.unfoldAsync (nextCmdSnap m .  cancelTk (←read)) initSnap (some fun _ => pure TaskError.aborted)
+  def unfoldCmdSnaps (m : DocumentMeta) (initSnap : Snapshot) (cancelTk : CancelToken) : ServerM (AsyncList ElabTaskError Snapshot) := do
+    AsyncList.unfoldAsync (nextCmdSnap m . cancelTk (←read)) initSnap
 
   /-- Compiles the contents of a Lean file. -/
   def compileDocument (m : DocumentMeta) : ServerM Unit := do
-    let headerSnap@⟨_, _, _, SnapshotData.headerData env msgLog opts⟩ ← Snapshots.compileHeader m.text.source
+    let headerSnap@{ data := SnapshotData.headerData cmdState, .. } ← Snapshots.compileHeader m.text.source
       | throwServerError "Internal server error: invalid header snapshot"
+    let opts : Options := {}
+    let opts := opts.setBool `interpreter.prefer_native false
```

Comment:
Why do we need this?

Reply:
Thank you for reminding me about this. The interpreter updates correctly without this for the usual `def`s, but I found that sometimes with custom `elab` commands it would actually get stuck using the native version, so I re-added this hack. I can try to triage this later or remove the option if it does not reproduce anymore.